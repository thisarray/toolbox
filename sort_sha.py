"""Sort the checksums found in the file at path case insensitively."""

if __name__ == '__main__':
    import argparse
    import os.path
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument(
        'path', nargs='?', default='',
        help='path to the checksum file')
    args = parser.parse_args()

    if os.path.isfile(args.path):
        buffer = []
        with open(args.path, 'r', encoding='utf-8') as f:
            for line in f:
                cleaned = line.strip()
                if len(cleaned) <= 0:
                    continue
                fields = cleaned.split(maxsplit=1)
                if len(fields) < 2:
                    continue
                if fields[1].startswith('*'):
                    # Remove asterisk in the checksum file
                    filename = fields[1].lstrip('*')
                else:
                    filename = fields[1]
                buffer.append((filename.lower(), filename, fields[0]))
        buffer.sort()
        for lowered, filename, checksum in buffer:
            print('{}  {}'.format(checksum, filename))
