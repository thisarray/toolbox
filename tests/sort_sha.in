abc  foo.txt
def  BAZ.txt
ghi *Bar.txt
