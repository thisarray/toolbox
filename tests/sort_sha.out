ghi  Bar.txt
def  BAZ.txt
abc  foo.txt
